src/nanoarrow/meson.build

```diff
@@ -147,4 +166,38 @@ if get_option('tests')
                                        include_directories: incdir)
   test('c_data_integration test', c_data_integration_test)
 
+  if get_option('ipc')
+      zlib_dep = dependency('zlib')
+      ipc_test_files = {
+          'nanoarrow-ipc-decoder': {
+              'deps': [nanoarrow_dep, flatcc_dep, arrow_dep, gtest_dep],
+          },
+          'nanoarrow-ipc-reader': {
+              'deps': [nanoarrow_dep, flatcc_dep, arrow_dep, gtest_dep],
+          },
+          'nanoarrow-ipc-files': {
+              'deps': [
+                  nanoarrow_dep,
+                  flatcc_dep,
+                  zlib_dep,
+                  arrow_dep,
+                  gtest_dep,
+                  nlohmann_json_dep
+              ],
+          },
+          'nanoarrow-ipc-hpp': {
+              'deps': [nanoarrow_dep, flatcc_dep, gtest_dep],
+          },
+      }
+
+      foreach name, config : ipc_test_files
+          exc = executable(
+              name + '-test',
+              name.replace('-', '_') + '_test.cc',
+              link_with: nanoarrow_ipc_lib,
+              dependencies: config['deps']
```

Comment:
Not sure I understand the first question - what problem are you trying to solve by making that a direct link argument? The way the dependency is declared (see [wrapdb source](https://github.com/mesonbuild/wrapdb/blob/a67b90d64e63fd52d0026ec24cb9972b831c2564/subprojects/packagefiles/gtest/googletest/meson.build#L14)) I believe Meson will take care of that linkage for you, alongside having the proper include and threads dependency

As far as the second question goes the way I had this previously the dependencies were not transitive, but I've since updated the nanoarrow_ipc_dep to include those:

```python
nanoarrow_ipc_dep = declare_dependency(..., dependencies: [nanoarrow_dep, flatcc_dep])
```

Reply:
I think when coming from CMake it is helpful to understand that the word "dependency" means something else in Meson. In CMake, dependency refers to "something that needs to be built before this target can be". In Meson, a dependency helps you transitively use include directories / link targets. You declare this via the `declare_dependency` function:

```python
base_dep = declare_dependency(
    include_directories: ['foo_dir'],
    link_with: [threads_lib],
)
```

So when you end up using that dependency on another target, it will transitively carry over the include / link flags for you, i.e.

```python
child_lib = library('child', dependencies: [base_dep])
```

expands into

```python
child_lib = library('child',
    include_directories: ['foo_dir'],
    link_with: [threads_lib],
)
```

(at least as far as I understand things)